iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/CreateCouplingDevice.java

```diff
@@ -109,51 +109,73 @@ public void apply(Network network, boolean throwException, ComputationManager co
-                    }
-                } else {
-                    LOGGER.warn("No busbar section position extension found on {}, only one disconnector is created.", bbs2.getId());
-                    noBusbarSectionPositionExtensionReport(reporter, bbs2);
-                }
-            } else {
-                LOGGER.warn("No busbar section position extension found on {}, only one disconnector is created.", bbs1.getId());
-                noBusbarSectionPositionExtensionReport(reporter, bbs1);
-            }
+            applyOnBusbarSections(voltageLevel1, voltageLevel2, bbs1, bbs2, reporter);
         }
         LOGGER.info("New coupling device was added to voltage level {} between {} and {}", voltageLevel1.getId(), busOrBbs1, busOrBbs2);
         newCouplingDeviceAddedReport(reporter, voltageLevel1.getId(), busOrBbsId1, busOrBbsId2);
     }
 
+    /**
+     * Apply the modification on the two specified busbar sections
+     */
+    private void applyOnBusbarSections(VoltageLevel voltageLevel1, VoltageLevel voltageLevel2, BusbarSection bbs1, BusbarSection bbs2, Reporter reporter) {
+        // busbar sections exist: voltage level is NODE_BREAKER
+        int breakerNode1 = voltageLevel1.getNodeBreakerView().getMaximumNodeIndex() + 1;
+        int breakerNode2 = breakerNode1 + 1;
+        int nbOpenDisconnectors = 0;
+
+        // Breaker
+        createNBBreaker(breakerNode1, breakerNode2, "", switchPrefixId, voltageLevel1.getNodeBreakerView(), false);
+
+        // Disconnectors
+        BusbarSectionPosition position1 = bbs1.getExtension(BusbarSectionPosition.class);
+        BusbarSectionPosition position2 = bbs2.getExtension(BusbarSectionPosition.class);
+        boolean checkAndFilterOtherBbs = position1 != null && position2 != null && position1.getSectionIndex() == position2.getSectionIndex();
+        if (position1 != null) {
+            // List of the bars for the first section
+            List<BusbarSection> bbsList1 = voltageLevel1.getNodeBreakerView().getBusbarSectionStream()
+                .filter(b -> b.getExtension(BusbarSectionPosition.class) != null)
+                .filter(b -> b.getExtension(BusbarSectionPosition.class).getSectionIndex() == position1.getSectionIndex()).collect(Collectors.toList());
+
+            // If the two busbarsections are in the same section, filter the second one if there are two busbarsections in the list
+            if (checkAndFilterOtherBbs && bbsList1.size() == 2) {
+                bbsList1 = bbsList1.stream().filter(b -> !b.getId().equals(busOrBbsId2)).collect(Collectors.toList());
+            }
+
+            // Disconnectors on side 1
+            createDisconnectorTopologyFromBusbarSectionList(voltageLevel1, breakerNode1, switchPrefixId, bbsList1, bbs1);
+            nbOpenDisconnectors += bbsList1.size() - 1;
+        } else {
+            createDisconnectorTopologyFromBusbarSectionList(voltageLevel1, breakerNode1, switchPrefixId, bbs1);
```

Comment:
This name had been given since it's the same as the method called inside (this method only creates an empty list and calls the "parent" method). But I'll rename it to make it more understandable.

Reply:
Actually you could also use createDisconnectorTopologyFromBusbarSectionList with the singletonList as argument